client/ayon_core/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -25,34 +25,50 @@ class SubstanceLoadProjectMesh(load.LoaderPlugin):
     icon = "code-fork"
     color = "orange"
 
-    options = [
-        qargparse.Boolean(
-            "preserve_strokes",
-            default=True,
-            help="Preserve strokes positions on mesh.\n"
-                 "(only relevant when loading into existing project)"
-        ),
-        qargparse.Boolean(
-            "import_cameras",
-            default=True,
-            help="Import cameras from the mesh file."
-        )
-    ]
-
-    def load(self, context, name, namespace, data):
+    @classmethod
+    def get_options(cls, contexts):
+        return [
+            BoolDef("allow_user_setting",
+                    default=True,
+                    label="Allow User Setting",
+                    tooltip=("Allow user to set up the project"
+                             " by their own\n")),
+            BoolDef("preserve_strokes",
+                    default=True,
+                    label="Preserve Strokes",
+                    tooltip=("Preserve strokes positions on mesh.\n"
+                             "(only relevant when loading into "
+                             "existing project)")),
+            BoolDef("import_cameras",
+                    default=True,
+                    label="Import Cameras",
+                    tooltip="Import cameras from the mesh file."
+            ),
+            EnumDef("texture_resolution",
+                    items=[128, 256, 512, 1024, 2048, 4096],
```

Comment:
With this PR coming up: https://github.com/ynput/ayon-core/pull/407 - I'd expect the project to also be able to go up to 8k. Are we missing an option here?

Reply:
Yes, but I just make a separate PR for that.

Reply:
But that's for the export size, not the project size - right?

Reply:
> But that's for the export size, not the project size - right?

yes it's different maybe I elaborate the name a little bit

Reply:
Anyway. So should this PR include a higher project resolution in this `items` list? I might be misunderstanding from your reply why it shouldn't.
```suggestion
                    items=[128, 256, 512, 1024, 2048, 4096, 8192],
```

Reply:
> Anyway. So should this PR include a higher project resolution in this `items` list? I might be misunderstanding from your reply why it shouldn't.

But in the texture configuration setting, it doesn't have the 8k supports, it is only limited to the texture export.

Reply:
Interesting - why would you do an 8k export for something that's based on 4k maps? What am I missing?

Reply:
> Interesting - why would you do an 8k export for something that's based on 4k maps? What am I missing?

there is an option for the texture export. It's in the export setting.
![image](https://github.com/ynput/ayon-core/assets/64118225/0128c145-bd4c-4214-af49-9a2c2aa6f25e)

Reply:
@moonyuet  the ```Project Config``` pop up window shows just for a few milisecs and dissappears :(

@BigRoy SPainter can do recalculation when exporting the 8K even tho the project being just 4K (they mention it in the docs - so it upres the output really)

Reply:
> @BigRoy SPainter can do recalculation when exporting the 8K even tho the project being just 4K (they mention it in the docs - so it upres the output really)

You just got to love software - well I guess that 'explains' it then.